src/main/java/emissary/admin/PlaceStarter.java

```diff
@@ -107,48 +108,52 @@ public static IServiceProviderPlace createPlace(final String theLocation, final
             shutdownFailedPlace(bindKey, null);
             return null; // couldn't start the place.
         }
 
         final long t2 = System.currentTimeMillis();
 
-        logger.debug("Started " + theLocation + " in " + (t2 - t1) / 1000.0 + "s");
+        if (logger.isDebugEnabled()) {
+            logger.debug("Started {} in {}s", theLocation, (t2 - t1) / 1000.0);
+        }
         return thePlace;
     }
 
     public static void shutdownFailedPlace(final String loc, final IServiceProviderPlace place) {
-
         try {
-            logger.warn("shutting down the failed place: " + loc);
+            logger.warn("shutting down the failed place: {}", loc);
             if (place != null) {
                 place.shutDown();
             } else {
                 // Force keys to be deregistered if we can
-                try {
-                    final IDirectoryPlace localDir = DirectoryPlace.lookup();
-                    final List<DirectoryEntry> entries = localDir.getMatchingEntries("*." + loc);
-                    if (entries != null && entries.size() > 0) {
-                        final List<String> keys = new ArrayList<String>();
-                        for (final DirectoryEntry entry : entries) {
-                            keys.add(entry.getKey());
-                        }
-                        logger.info("Forcing removal of " + keys.size() + " keys due to failed " + loc);
-                        localDir.removePlaces(keys);
-                    } else {
-                        logger.debug("Failed " + loc + " did not have any directory keys registered");
-                    }
-                } catch (EmissaryException ee) {
-                    logger.debug("NO local directory, cannot force key dereg for " + loc);
-                }
+                deregisterPlace(loc);
             }
             Namespace.unbind(loc);
         } catch (Throwable tt) {
-            logger.error("whoa there pardner... " + loc, tt);
+            logger.error("whoa there pardner... {}", loc, tt);
         }
     }
 
+    public static void deregisterPlace(final String loc) {
+        try {
+            final IDirectoryPlace localDir = DirectoryPlace.lookup();
+            final List<DirectoryEntry> entries = localDir.getMatchingEntries("*." + loc);
+            if (entries != null && !entries.isEmpty()) {
+                final List<String> keys = new ArrayList<>();
+                for (final DirectoryEntry entry : entries) {
+                    keys.add(entry.getKey());
+                }
+                logger.info("Forcing removal of {} keys due to failed {}", keys.size(), loc);
+                localDir.removePlaces(keys);
+            } else {
+                logger.debug("Failed {} did not have any directory keys registered", loc);
+            }
+        } catch (EmissaryException ee) {
+            logger.debug("NO local directory, cannot force key dereg for {}", loc);
```

Comment:
Should this and the debug message above be a higher log level? Something we would want to know about in production environment?

Reply:
This is a failure scenario, so the place the failed to start probably logged a stacktrace already. My guess is that this was turned down to debug because it is just overly noisy. We are trying to deregister the place, so we most likely don't need a bunch of stacktraces that we are having issues with deregistering the place that failed to start up.  We can set this to error if you want, but my suggestion would be to submit a separate pr for it. Thoughts?